Use pandoc's markdown image link text as short caption in LaTeX

bwhelm opened the pull request:
Use pandoc's markdown image link text as short caption in LaTeX

```
![Long caption](figure.jpg "short caption"){#fig:ref}
```

should produce in LaTeX:

```
\begin{figure}[htbp]
\centering
\includegraphics{figure.jpg}
\caption[short caption]{Long caption}
\label{fig:ref}
\end{figure}
```

In addition, this adds the link text to html, html5, and markdown formats.

**Notes**
- I'm not sure why the math is coming out different in my versions of spec.html and spec.html5. That would seem to be unaffected by my changes here.
- I'm not sure how to pretty print json the same way as the original, so my spec.json has too many changes (including changes to the Unicode text that happened when I prettified it). That should be fixed.
- More important: I don't see the link text showing up in the json output when run through this filter. I'm not sure how to change that.
- Also important: if there are no figure attributes, the link text doesn't show up in the output. I'm happy with that, but it is a limitation worth noting.

bwhelm committed:
Use pandoc's markdown image link text as short caption in LaTeX

    ![Long caption](figure.jpg "short caption"){#fig:ref}

should produce in LaTeX:

    \begin{figure}[htbp]
    \centering
    \includegraphics{figure.jpg}
    \caption[short caption]{Long caption}
    \label{fig:ref}
    \end{figure}

In addition, this adds the link text to html, html5, and markdown formats.

*Notes*
- I'm not sure why the math is coming out different in my versions of spec.html and spec.html5. That would seem to be unaffected by my changes here.
- I'm not sure how to pretty print json the same way as the original, so my spec.json has too many changes. That should be fixed.

aaren commented:
R.e. math: The test references are produced by `pandoc spec.md --to format --mathjax`. Maybe you're missing the `mathjax`?

aaren commented:
R.e. json: I feed this through [jq](https://stedolan.github.io/jq/)

```
pandoc spec.md --to json --mathjax | jq . > spec.json
```

aaren commented:
I've not come across `short caption` before. Does it get used in figure lists?

bwhelm committed:
Corrections to test documents.

bwhelm commented:
Re math: You're right. I was not using the '--mathjax' flag. Corrected now.

Re json: Thanks. I used jq's online tool to correct the output.

Re short caption: Yes, the short caption shows up in a list of figures. This is analogous to optional chapter/section titles: `\section[short title]{long title}` means that "short title" shows up in the table of contents (and running heads), but "long title" shows up in the main text.

bwhelm committed:
Fixes problems for short captions that arise in 2 conditions:

(1) when figure title is empty -- when we have a figure of
the form:

    [Caption](image.jpg){#fig:ref}

In this case, the earlier code produces:

    \caption[]{Caption}

and the captions package would then print out the '[]'. This avoids
that.

(2) when the figure is unnumbered: then having a short
caption is senseless. The earlier code would produce:

    \caption*[short caption]{Caption}

and the captions package would include the '[short caption'] along with
the caption. Again, this avoids this problem.

bwhelm committed:
Don't want brackets showing up around short caption for formats other
than LaTeX.
